docs/config/advanced.md

```diff
@@ -82,14 +82,74 @@ Filters: there are several operators that act as filters to remove results you d
 
 - `fix` allows for an expression to be displayed to the user as the suggested autofix. Can be applied with `--autofix`
 
+## Path Filtering
+
+### Excluding a Rule in Paths
+
+To ignore a specific rule on specific files, set a `paths:` key with one or more filters like so:
+
+```yaml
+rules:
+  - id: eqeq-is-bad
+    pattern: $X == $X
+    paths:
```

Comment:
@underyx Thank you for the documentation! Did you consider keys like `exclude` or `include-only`, similar to Zulip's existing terminology and syntax? See https://github.com/zulip/zulip/blob/master/tools/linter_lib/custom_check.py#L7-L13

Similarly, did you consider and decide against accepting a list of paths?

```yaml
rules:
   exclude: ["*.jinja2", "_test.go"]
```

Reply:
Yep, I decided against.

- Whichever we choose, customers will have to figure out the intricacies of the config syntax, such as whether items in a list are AND or OR-concatenated. Thought it'd be nice if we save a couple brain cycles for users by just reusing the exact same concepts as for patterns. Although paths is more limited currently and doesn't support stuff like `path-either`, which might be surprising.
- Being able to specify a `filename-not` vs. `path-not` is useful when you don't want to figure out the implementation details of the glob matching, such as how you match all python files in the root of the repo, or all python files in all subdirectories. with the simple exclude list you'd need to know to write `**/*.py`.

Reply:
fwiw having an exclude or include list is a pretty common pattern in many tools and it's pretty understood that if any file matches an item in an exclude list it is ignored.

Reply:
Right, that was a bad example to pick. Should I change the config syntax?

Reply:
but since this is experimental anyway ill leave up to you and @DrewDennison